source/app/controllers/urls_controller.rb

```diff
@@ -1,2 +1,81 @@
 class UrlsController < ApplicationController
+  before_action :set_url, only: [:show, :edit, :update, :destroy]
+
+  # GET /urls
+  # GET /urls.json
+  def index
+    @urls = Url.all
+  end
+
+  # GET /urls/1
+  # GET /urls/1.json
+  def show
+  end
+
+  # GET /urls/new
+  def new
+    @url = Url.new
+  end
+
+  # GET /urls/1/edit
+  def edit
+  end
+
+  def goto
+    id = Url.decode_shortcode params[:shortcode]
+    url = Url.find id
+    url.click_count += 1
```

Comment:
this works here, but we normally try to push this logic into the model if possible. for example...
```ruby
class Url
  def increment
    self.click_count += 1
    save!
  end
end
```